sensing/src/main/java/com/google/android/sensing/capture/CaptureRequest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.sensing.capture
+
+import android.media.MediaRecorder
+import java.util.UUID
+
+/**
+ * Base class representing a capture request with common properties.
+ *
+ * @param externalIdentifier App identifier for this capture request.
+ * @param outputFolder The destination folder where captured output will be saved.
+ * @param outputFormat The output format (e.g., "jpeg", "video/mp4v-es")
+ * @param outputTitle Label to display in place of the data.
+ */
+sealed class CaptureRequest(
+  open val externalIdentifier: String,
+  open val outputFolder: String,
+  open val outputFormat: String,
+  open val outputTitle: String,
+  val captureId: String = UUID.randomUUID().toString(),
+)
+
+sealed class CameraCaptureRequest(
+  override val externalIdentifier: String,
+  override val outputFolder: String,
+  override val outputFormat: String,
+  override val outputTitle: String,
```

Comment:
do you have to declare these if they're not actually being overriden?

Reply:
But they need to be overriden.
I dont get your question.